modules/ossm-threescale-istio-adapter-troubleshooting.adoc

```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v1x/threescale_adapter/threescale-adapter.adoc
+// * service_mesh/v2x/threescale_adapter/threescale-adapter.adoc
+
+[id="ossm-threescale-istio-adapter-troubleshooting_{context}"]
+= 3scale Istio adapter troubleshooting
+
+You might want to check whether the 3scale Istio adapter is working as expected. If this is not the case, you can troubleshoot from the list of scenarios.
+
+Perform the following steps to check that the adapter is working correctly:
+
+. Ensure the _3scale-adapter_ pod is running in the control plane namespace, that is `istio-system`.
+. Check that the _3scale-adapter_ pod has printed out information about itself booting up, such as its version.
+. When performing requests to the services protected by the 3scale adapter integration, always try requests that lack the right credentials and ensure they fail. Check out the 3scale adapter logs to gather additional information.
+
+If the steps above do not help, see the following list, which you can use to troubleshoot the 3scale Istio adapter.
+
+== Troubleshooting checklist
+
+There are a number of scenarios that might be causing your 3scale Istio adapter integration to not function properly:
+
+* Forgot to label the service workloads with the `service-mesh.3scale.net/credentials` key.
+* Forgot to label the service workloads with `service-mesh.3scale.net/service-id` when using handlers that do not contain a `service_id` so they are reusable per account.
+* The _Rule_ custom resource points to the wrong handler or instance custom resources, or the references lack the corresponding namespace suffix.
+* The _Rule_ custom resource match section cannot possibly match the service you are configuring, or it points to a destination workload that is not currently running or does not exist.
+* Wrong access token or URL for the 3scale Admin Portal in the handler.
+* The _Instance_ custom resource's `params/subject/properties` section fails to list the right parameters for `app_id`, `app_key`, or `client_id`, either because they specify the wrong location such as the query parameters, headers, and authorization claims, or the parameter names do not match the requests used for testing.
+* Trying and failing to use the configuration generator without realizing that it actually lives in the adapter container image and needs `oc exec` to invoke it.
```

Comment:
As explained [in the ticket](https://issues.redhat.com/browse/THREESCALE-6894?focusedCommentId=16387012&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16387012), I'd prefer that this list also includes a reference to the check the YAML format/indentation and any missing sections, even if it's not explicitly listed as a scenario by itself. Reminding the user to check the YAML is well indented and has no missing sections _along pointing them to the CLI tool for automation of the process_ is going to be a good service for support people and myself when users forget about these basics and in desperation open new support cases.